src/components/AddComment.jsx

```diff
@@ -0,0 +1,76 @@
+import { useState } from "react";
+import { useParams } from "react-router-dom";
+import { postComment } from "../api";
+
+const AddComment = ({ err, setErr, setComments }) => {
+  const { article_id } = useParams();
+  const [body, setBody] = useState("");
+  const [username, setUsername] = useState("");
+  const [buttonDisable, setButtonDisable] = useState(false);
+
+  const handlePostComment = (event) => {
+    event.preventDefault();
+    setButtonDisable(true);
+    const newComment = {
+      username: username,
+      body: body,
+      votes: 0,
+    };
+
+    setBody([]);
+    setUsername("");
+
+    if (body.length === 0) {
+      setErr("No Information! Please try again!");
+      setButtonDisable(false);
+    } else {
+      postComment(article_id, newComment)
+        .then((addedComment) => {
+          setComments((currentComments) => {
+            setButtonDisable(false);
+            setErr(null);
+            return [addedComment, ...currentComments];
+          });
+        })
+        .catch((err) => {
+          setErr("No Information! Please try again!");
```

Comment:
Only issue with this is i get the same message if i try and post offline. A bit confusing message - might just be better as `something went wrong`